handlers/utils/utils.go

```diff
@@ -141,11 +118,17 @@ func readBuffer(reader io.ReadCloser) string {
 }
 
 func isStrSet(m map[string]interface{}, key string) bool {
-	return m[key] != nil && len(m[key].([]string)) > 0
+	ok := false
+	switch m[key].(type) {
+	case string:
+		ok = len(InterfaceToString(m[key])) > 0
+	case []string:
+		ok = len(m[key].([]string)) > 0
+	}
+	return m[key] != nil && ok
 }
 
-// this method check if a field exists in a map
-func getFieldsIfExist(m map[string]interface{}, fields ...string) (interface{}, bool) {
+func GetFieldsIfExist(m map[string]interface{}, fields ...string) (interface{}, bool) {
```

Comment:
default means it is neither map nor nil otherwise it will fall into the previous two cases in switch